pkg/kubectl/scale.go

```diff
@@ -177,3 +170,16 @@ func scaleHasDesiredReplicas(sClient scaleclient.ScalesGetter, gr schema.GroupRe
 			desiredReplicas == actualScale.Status.Replicas, nil
 	}
 }
+
+// WaitForScaleHasDesiredReplicas waits until condition scaleHasDesiredReplicas is satisfied
+// or returns error when timeout happens
+func WaitForScaleHasDesiredReplicas(sClient scaleclient.ScalesGetter, gr schema.GroupResource, resourceName string, namespace string, newSize uint, waitForReplicas *RetryParams) error {
+	err := wait.PollImmediate(
```

Comment:
If waitForReplicas is nil, it will crash.

if waitForReplicas != nil should be handled here.

Reply:
done